source/extensions/filters/http/jwt_authn/jwks_cache.cc

```diff
@@ -92,9 +111,13 @@ class JwksDataImpl : public JwksCache::JwksData, public Logger::Loggable<Logger:
   TimeSource& time_source_;
   // The pubkey expiration time.
   MonotonicTime expiration_time_;
+  // Jwt object for verified token.
+  std::unique_ptr<::google::jwt_verify::Jwt> jwt_;
```

Comment:
The cache should be

// Map a token to its result in the cache
std::map<token,  TokenResult> token_cache_;

Struct TokenResult {
     // cache expire time, default cache for 5 minutes
     MonotonicTime expiration_time_;

    // If this is value,  this token is good,  stored its parsed jwt
    optional<Jwt>   good_jwt_;

    // if token verification fails, stores the failed status
    optional<Status>  failed_jwt_status_;
}

In the jwt_authn config, in the provider config, add

   // enable cache 
   bool enable_token_cache ;

  // specify token cache during,  default to 5 minutes
   Duration  token_cache_duration;

Reply:
should the cache have a (optionally) limited size to cap the amount of memory it can use?

Reply:
ahh nvm, i saw you commented on that.

Reply:
Done.